chain/epoch-manager/src/shard_tracker.rs

```diff
@@ -127,8 +152,15 @@ impl ShardTracker {
                 return true;
             }
         }
-        matches!(self.tracked_config, TrackedConfig::AllShards)
-            || self.tracks_shard(shard_id, parent_hash).unwrap_or(false)
+        match self.tracked_config {
+            TrackedConfig::AllShards => {
+                // Avoid looking up EpochId.
+                true
+            }
+            _ => {
+                self.tracks_shard_next_epoch_from_prev_block(shard_id, parent_hash).unwrap_or(false)
```

Comment:
Note that the previous implementation of `will_care_about_shard` wasn't looking at the next epoch. It happened to work because all possible `TrackedConfig` values were true or false for all epochs.